src/poetry/utils/env.py

```diff
@@ -421,9 +417,7 @@ class EnvCommandError(EnvError):
     def __init__(self, e: CalledProcessError, input: Optional[str] = None) -> None:
         self.e = e
 
-        message = "Command {} errored with the following return code {}, and output: \n{}".format(
-            e.cmd, e.returncode, decode(e.output)
-        )
+        message = f"Command {e.cmd} errored with the following return code {e.returncode}, and output: \n{decode(e.output)}"
```

Comment:
Another side note: I'm not sure `_compat.decode` or `_compat.encode` serve any useful function anymore. Certainly, attempting to decode with multiple encodings and then ignoring Unicode errors strikes me as a bit reckless. An issue discussing them or a PR removing them are both welcome.